core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -195,14 +195,16 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     dataSource.inTransaction { dataAccess =>
       withWorkspaceContext(workspaceName, dataAccess) { workspaceContext =>
         getMaximumAccessLevel(RawlsUser(userInfo), workspaceContext, dataAccess) flatMap { accessLevel =>
-          if (accessLevel < WorkspaceAccessLevels.Read)
-            DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.NotFound, noSuchWorkspaceMessage(workspaceName))))
-          else {
-            for {
-              stats <- getWorkspaceSubmissionStats(workspaceContext, dataAccess)
-              owners <- getWorkspaceOwners(workspaceContext.workspace, dataAccess)
-            } yield {
-              RequestComplete(StatusCodes.OK, WorkspaceListResponse(accessLevel, workspaceContext.workspace, stats, owners))
+          getUserSharePermissions(workspaceContext, accessLevel, dataAccess) flatMap { canShare =>
+            if (accessLevel < WorkspaceAccessLevels.Read)
+              DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.NotFound, noSuchWorkspaceMessage(workspaceName))))
+            else {
+              for {
+                stats <- getWorkspaceSubmissionStats(workspaceContext, dataAccess)
+                owners <- getWorkspaceOwners(workspaceContext.workspace, dataAccess)
+              } yield {
+                RequestComplete(StatusCodes.OK, WorkspaceResponse(accessLevel, canShare, workspaceContext.workspace, stats, owners))
```

Comment:
I changed this response type because it was also used in the listWorkspaces API call. I needed to return the canShare value but calculating it for the entire list of workspaces was useless and would add a lot of extra queries. So I broke them apart